### ISSUE TITLE
Heuristic dissector

### ISSUE THREAD

**Issue by expploitt** [cguimaraes/zenoh-dissector#14]
Hello @cguimaraes,

One of the main problems I found in the zenoh dissector was that you sniff as zenoh messages only the ones which uses the 7447 port. 

I'm working in the use of a heuristic dissector for TCP sniff purpose. I forked your repo in order to work with this and try to reuse the decoding lib that you code .

**Comment by cguimaraes** [comment]
Hi @expploitt ,
7447 (both for UDP and TCP) are the default ports used by Zenoh. 
In any case, if you are using a different set of ports, you can still decode them in Wireshark by doing the following:

> `Right click on a packet and select "Decode as...". Then choose "Zenoh" in the list.`

**Comment by expploitt** [comment]
Yes, but not always, i.e. the UDP scout discovery is using port 7447 but then the HELLO response use another port.

After UDP SCOUT phase, the peers establishes an TCP connection with random ports, in order to not block port 7447. This TCP messages are not detected as zenoh messages because of the ports.

In the last PCAP I attached in another issue you can see that some TCP messages are from zenoh but they are not detected for that reason.

**Comment by expploitt** [comment]
I added an udp heuristic checker to detect the HELLO messages

![imagen](https://user-images.githubusercontent.com/75727913/119320043-6f587700-bc7b-11eb-922f-8e4cbc74079b.png)

However, this method introduces some false positives. Would be a better approach to save the src_port of the SCOUT messages and analyze the messages using that port?
